### PR TITLE
[New Rule ] Incoming DCOM Lateral Movement with MSHTA

### PR DESCRIPTION
**Pull request by Samirbous**
## Issues

Resolves https://github.com/elastic/detection-rules/issues/458

## Contributor checklist

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)?

### CODE REVIEW

**Comment by Samirbous** [comment]
Example of Alert:

image

File: rules/windows/lateral_movement_dcom_hta.toml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+ecs_version = ["1.6.0"]
+maturity = "production"
+updated_date = "2020/11/03"
+
+[rule]
+author = ["Elastic"]
+description = """
+Identifies use of Distributed Component Object Model (DCOM) to run commands from a remote host, which are launched via
+the HTA Application COM Object. This behavior may indicate an attacker abusing a DCOM application to stealthy move
+laterally.
```
Comment (threat-punter) [inline rules/windows/lateral_movement_dcom_hta.toml:12]:
```suggestion
Identifies the use of Distributed Component Object Model (DCOM) to execute commands from a remote host, which are
launched via the HTA Application COM Object. This behavior may indicate an attacker abusing a DCOM application to move
laterally while attempting to evading detection.
```

**Review by threat-punter — APPROVED** [review]
👍

File: rules/windows/lateral_movement_dcom_hta.toml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+query = '''
+sequence with maxspan=1m
+  [process where event.type in ("start", "process_started") and
+   process.name == "mshta.exe" and process.args == "-Embedding"
+  ] by host.id, process.entity_id
+  [network where event.type == "start" and process.name=="mshta.exe" and 
+   event.action == "connection_accepted" and network.direction == "incoming" and network.transport == "tcp" and
+   source.port > 49151 and destination.port > 49151
+  ] by host.id, process.entity_id
+'''
```
Comment (brokensound77) [inline rules/windows/lateral_movement_dcom_hta.toml:35]: `event.action` would prevent this from triggering with winlogbeat here, but I think the logic should be fine without it

```suggestion
query = '''
sequence with maxspan=1m
  [process where event.type in ("start", "process_started") and
     process.name : "mshta.exe" and process.args : "-Embedding"
  ] by host.id, process.entity_id
  [network where event.type == "start" and process.name : "mshta.exe" and 
     network.direction == "incoming" and network.transport == "tcp" and
     source.port > 49151 and destination.port > 49151
  ] by host.id, process.entity_id
'''
```

Comment (Samirbous) [reply rules/windows/lateral_movement_dcom_hta.toml:35]: its ok as long as the incoming direction and both ports are higher than 49151 preserved: 

image

File: rules/windows/lateral_movement_dcom_hta.toml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+description = """
+Identifies the use of Distributed Component Object Model (DCOM) to execute commands from a remote host, which are
+launched via the HTA Application COM Object. This behavior may indicate an attacker abusing a DCOM application to move
+laterally while attempting to evading detection.
+"""
+from = "now-9m"
+index = ["winlogbeat-*", "logs-endpoint.events.*"]
+language = "eql"
+license = "Elastic License"
+name = "Incoming DCOM Lateral Movement with MSHTA"
```
Comment (bm11100) [inline rules/windows/lateral_movement_dcom_hta.toml:17]:
```suggestion
name = "Incoming DCOM Lateral Movement via MSHTA"
```

**Review by bm11100 — APPROVED** [review]

**Pull request merged**